crates/turborepo-lib/src/turbo_json/parser.rs

```diff
@@ -0,0 +1,576 @@
+use std::{
+    collections::BTreeMap,
+    fmt::{Debug, Display},
+    sync::Arc,
+};
+
+use biome_deserialize::{
+    json::deserialize_from_json_str, Deserializable, DeserializableValue,
+    DeserializationDiagnostic, DeserializationVisitor, Text, VisitableType,
+};
+use biome_diagnostics::DiagnosticExt;
+use biome_json_parser::JsonParserOptions;
+use biome_json_syntax::TextRange;
+use miette::{Diagnostic, SourceSpan};
+use thiserror::Error;
+
+use crate::{
+    cli::OutputLogsMode,
+    config::ConfigurationOptions,
+    run::task_id::TaskName,
+    turbo_json::{Pipeline, RawTaskDefinition, RawTurboJson, SpacesJson, Spanned},
+};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("failed to parse turbo json")]
+#[diagnostic(code(turbo_json::parser::parse_error))]
+pub struct Error {
+    #[related]
+    diagnostics: Vec<ParseDiagnostic>,
+}
+
+struct BiomeMessage<'a>(&'a biome_diagnostics::Error);
+
+impl Display for BiomeMessage<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.description(f)
+    }
+}
+
+impl From<biome_diagnostics::Error> for ParseDiagnostic {
+    fn from(diagnostic: biome_diagnostics::Error) -> Self {
+        let location = diagnostic.location();
+        let message = BiomeMessage(&diagnostic).to_string();
+        Self {
+            message,
+            source_code: location
+                .source_code
+                .map(|s| s.text.to_string())
+                .unwrap_or_default(),
+            label: location.span.map(|span| {
+                let start: usize = span.start().into();
+                let end: usize = span.end().into();
+                let len = end - start;
+                (start, len).into()
+            }),
+        }
+    }
+}
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("{message}")]
+#[diagnostic(code(turbo_json::parser::parse_error))]
+struct ParseDiagnostic {
+    message: String,
+    #[source_code]
+    source_code: String,
+    #[label]
+    label: Option<SourceSpan>,
+}
+
+impl Deserializable for OutputLogsMode {
+    fn deserialize(
+        value: &impl DeserializableValue,
+        name: &str,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        match String::deserialize(value, name, diagnostics)?.as_str() {
+            "full" => Some(OutputLogsMode::Full),
+            "none" => Some(OutputLogsMode::None),
+            "hash-only" => Some(OutputLogsMode::HashOnly),
+            "new-only" => Some(OutputLogsMode::NewOnly),
+            "errors-only" => Some(OutputLogsMode::ErrorsOnly),
+            unknown_variant => {
+                const ALLOWED_VARIANTS: &[&str] =
+                    &["full", "none", "hash-only", "new-only", "errors-only"];
```

Comment:
I'm really concerned we're going to accidentally get these out of sync. Can we leverage [ValueEnum](https://docs.rs/clap/latest/clap/trait.ValueEnum.html) to generate this list instead of hard coding this?